### PR TITLE
[EXP] Enable CI

### LINKED ISSUE #42: Cannot use pytest
When importing pyraf from within pytest, I get the following error on Python 3:
```
==================================== ERRORS ====================================
_______________________ ERROR collecting debian/tests.py _______________________
tests.py:8: in <module>
    from pyraf import iraf
/usr/lib/python3/dist-packages/pyraf/__init__.py:126: in <module>
    iraf.Init(doprint=0, hush=1)
/usr/lib/python3/dist-packages/pyraf/iraffunctions.py:257: in Init
    userpkg.run(_doprint=0, _hush=hush, _save=1)
/usr/lib/python3/dist-packages/pyraf/iraftask.py:359: in run
    self._run(redirKW, specialKW)
/usr/lib/python3/dist-packages/pyraf/iraftask.py:1667: in _run
    ???
/usr/lib/python3/dist-packages/pyraf/iraftask.py:1470: in _runCode
    ???
<CL script clpackage.user>:28: in login
    Pipe1 = iraf.cl(Stdin=Pipe2, Stdout=1)
/usr/lib/python3/dist-packages/pyraf/iraftask.py:767: in __call__
    return self.run(*args, **kw)
/usr/lib/python3/dist-packages/pyraf/iraftask.py:359: in run
    self._run(redirKW, specialKW)
/usr/lib/python3/dist-packages/pyraf/iraftask.py:1202: in _run
    self._pyFunction(*pl)
/usr/lib/python3/dist-packages/pyraf/iraffunctions.py:2485: in _clProcedure
    clExecute(_sys.stdin.read(), locals=theLocals, Stdin=_sys.__stdin__)
/usr/lib/python3/dist-packages/pyraf/iraffunctions.py:3088: in clExecute
    exec(pycode.vars.proc_name+"(taskObj=iraf.cl)", locals)
<string>:1: in <module>
    ???
<CL script CL1>:12: in string_proc
    ???
/usr/lib/python3/dist-packages/pyraf/iraffunctions.py:2088: in clOscmd
    status = _subproc.subshellRedir(s, shell=shell)
/usr/lib/python3/dist-packages/pyraf/subproc.py:736: in subshellRedir
    return systemRedir((shell, "-c", cmd))
/usr/lib/python3/dist-packages/pyraf/subproc.py:718: in systemRedir
    process.run()
/usr/lib/python3/dist-packages/pyraf/subproc.py:790: in run
    sys.stderr.write(tostr(s))
/usr/lib/python3/dist-packages/stsci/tools/for2to3.py:82: in tostr
    return s.decode(encoding)
E   UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 56: ordinal not in range(128)
!!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!!
=========================== 1 error in 2.74 seconds ============================
```

On Python 2, it ~~also does not work~~ (#43) works well.

### LINKED ISSUE #44: Port tests from pandokia to this repo and possibly use Travis CI instead
Originated from a request by @olebole in https://github.com/spacetelescope/pyraf/issues/36#issuecomment-368487128

p.s. Some of the tests in pandokia might require IRAF. I don't think those are going to be ported over.

p.p.s. This is more painful than it sounds. Given the resources we have, this might actually never happen...

### PR DESCRIPTION
**Pull request by pllim**
For better or worse...
Fix #42 
Fix #44

- [x] Use Jenkins with proper IRAF installation?
- [x] Make sure they are not skipped. Need Joe's Docker magic.

- [x] Add tests from Ole

- [x] Move "embedded" tests in `lib/pyraf/*.py` modules to `tests` properly? After talking to Chris S., need to keep duplicate of these and deleted modules in a new dir `lib/pyraf/old_files` so he can run them his way.
- Port over from pandokia (strike-through means those tests are now redundant and thus not ported over):
  - [x] pyraf/cli/division/test
  - [x] pyraf/cli/sscanf/test
  - [x] pyraf/cli/whereis/test
  - [x] pyraf/cli/which/test
  - [x] pyraf/core/graphics_checks/checks
  - [x] pyraf/core/graphics_checks/checks.test_dumpspecs
  - [x] pyraf/core/irafparlist/test
  - [x] pyraf/core/subproc/test
  - [x] pyraf/gki_misc/decode/test
  - [x] pyraf/gki_misc/psikern/test
  - [x] pyraf/gki_misc/sys/test
  - [x] pyraf/invoke_redo/invoke
  - [x] pyraf/invoke_redo/invoke.test_cl_redir
  - [x] pyraf/invoke_redo/invoke.test_dashcommand
  - [x] pyraf/invoke_redo/invoke.test_dashcommand_simple
  - [x] pyraf/invoke_redo/invoke.test_dashi
  - [x] pyraf/invoke_redo/invoke.test_dashy
  - [x] ~~pyraf/invoke_redo/invoke.test_get_status_and_output_of~~
  - [x] ~~pyraf/invoke_redo/invoke.test_has_latest_testcode~~
  - [x] ~~pyraf/invoke_redo/invoke.test_nows_get_status_and_output_of~~
  - [x] ~~pyraf/invoke_redo/invoke.test_usage~~
  - [x] pyraf/using_tasks/207/test207
  - [x] pyraf/using_tasks/207/test207.test
  - [x] pyraf/using_tasks/pset_as_func_arg/arg2msstat
  - [x] pyraf/using_tasks/pset_as_func_arg/arg2msstat.test
  - [x] pyraf/using_tasks/pset_as_retval/msstat2egstp
  - [x] pyraf/using_tasks/pset_as_retval/msstat2egstp.test

Follow up after merging this:

- [ ] Open issue about addressing bugs found in some `xfail` ported tests (`grep` for "BUG" and "TODO").
- [ ] Disable pandokia tests for pyraf.
- [ ] Setup Jenkins nightly cron. We can skip cron if there has been a run on master in the past 24 hours (e.g., from push or merge commit).

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,101 @@
+// Obtain files from source control system.
+if (utils.scm_checkout()) return
+
+// Globals
+PIP_INST = "pip install"
+CONDA_INST = "conda install -y -q"
+PY_SETUP = "python setup.py"
+PYTEST_ARGS = "--basetemp=tests_output --junitxml results.xml"
```
Comment (jhunkeler) [inline Jenkinsfile]: How do you feel about adding `--capture=sys` or `--capture=fd` to `PYTEST_ARGS`? I'm noticing that IRAF tasks are inconsistently verbose, and PyRAF doesn't always redirect `stdout` and `stderr` streams correctly when instructed to.

Comment (pllim) [reply Jenkinsfile]: Why not? Let's do it. It's a wild wild West here.

Comment (pllim) [reply Jenkinsfile]: @jhunkeler , wait a minute. Now I remember that I already have `addopts = --capture=no` in `setup.cfg`. Is that not sufficient? Won't Jenkins be smart enough to pick up `pytest` options from there?

File: lib/pyraf/tests/utils.py [hunk 2]
```diff
@@ -4,8 +4,11 @@
 from astropy.utils.data import get_pkg_data_contents
 
 IS_PY2 = sys.version_info < (3, 0)
+DATA_DIR = os.path.join(os.path.dirname(__file__), 'data')
```
Comment (pllim) [inline lib/pyraf/tests/utils.py]: Why is this necessary? Can't you just use `get_pkg_data_filename` from `astropy.utils.data`? See L4.

**Comment by pllim** [comment]
Hey, hey, it's green. Merging.
